includes/wikia/traits/TitleTrait.php

```diff
@@ -0,0 +1,160 @@
+<?php
+/**
+ * A trait that let's you expand the base Title class
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ */
+
+trait TitleTrait {
+	// Required base-class methods
+	abstract function getNamespace();
+	abstract function getDBkey();
+
+	/**
+	 * Returns a list of pages that link to that Title
+	 *
+	 * @param int $level How deep should you go?
+	 * @return array
+	 */
+	public function getIndirectLinks( $level = 0 ) {
+		global $wgContentNamespaces;
+
+		$rows = [];
+
+		$dbr = wfGetDB( DB_SLAVE );
+		$options = [];
+
+		$hidelinks = false;
+		$hideredirs = false;
+		$hidetrans = false;
+		$hideimages = false; //$target->getNamespace() != NS_FILE;
+
+		$fetchlinks = ( !$hidelinks || !$hideredirs );
+
+		// Make the query
+		$plConds = [
+			'page_id=pl_from',
+			'pl_namespace' => $this->getNamespace(),
+			'pl_title' => $this->getDBkey(),
+		];
+		if( $hideredirs ) {
+			$plConds['rd_from'] = null;
+		} elseif( $hidelinks ) {
+			$plConds[] = 'rd_from is NOT NULL';
+		}
+
+		$tlConds = [
+			'page_id=tl_from',
+			'tl_namespace' => $this->getNamespace(),
+			'tl_title' => $this->getDBkey(),
+		];
+
+		$ilConds = [
+			'page_id=il_from',
+			'il_to' => $this->getDBkey(),
+		];
+
+		if ( is_array( $wgContentNamespaces ) && !empty( $wgContentNamespaces ) ) {
+			$namespaces = implode( ',', $wgContentNamespaces );
```

Comment:
As a best practice, prefer `DatabaseBase::makeList`, even if you trust the source.